### PR TITLE
Feature/unit test

### LINKED ISSUE #116: [tests] `actions/nations.js`
## Acceptance criteria
- [x] Add test's for `actions/nations.js`
- [ ] updated doc's (in code) what this file is about.

### LINKED ISSUE #117: [tests] actions/profile.js
## Acceptance criteria

* [ ] Add test's for `actions/profile.js`
* [ ] update what this is about

### LINKED ISSUE #147: [tests] src/reducers/nations.js
## Acceptance criteria
- [ ] Add test's
- [ ] Add doc's / readme

### LINKED ISSUE #148: [tests] src/reducers/profile.js
## Acceptance criteria
- [ ] Add test#s
- [ ] Add doc's

### LINKED ISSUE #155: [tests] src/sagas/nations.js
## Acceptance criteria
- [ ] Add test's
- [ ] Add documentation (jsdoc)

### LINKED ISSUE #156: [tests] src/sagas/profile.js
## Acceptance criteria
- [ ] Add test's
- [ ] add doc's (jsdoc)

### PR DESCRIPTION
**Pull request by SeniorMobileDeveloper**
#### Changes
* unit test for profile actions (closes #117)
* unit test for profile reducers (closes #148)
* unit test for nations actions (closes #116)
* unit test for nations reducers (closes #147)
* unit test for nations sagas (closes #155)
* unit test for profile sagas (closes #156)

### CODE REVIEW

**Comment by seland** [comment]
@SeniorMobileDeveloper Please include issues that will be closed by that PR in description.

**Comment by SeniorMobileDeveloper** [comment]
The following issues can be closed by this PR.
#116 
#117 
#147 
#148 
#155

**Comment by SeniorMobileDeveloper** [comment]
Just added unit test for profile saga thus the issue #156 can be closed too.

File: __tests__/src/reducers/nations.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	const state = reducer(initialState, joinNation())
+	expect(state).toEqual({...initialState, inProgress: true})
+  done()
+})
+
+test('reducer - leaveNation', (done) => {
+	const state = reducer(initialState, leaveNation())
+	expect(state).toEqual({...initialState, inProgress: true})
+  done()
+})
```
Comment (seland) [inline __tests__/src/reducers/nations.js]: Test for DONE_FETCH_NATIONS action is missing, please add.

File: __tests__/src/sagas/nations.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    nation: {
+      create: jest.fn(),
+      all: jest.fn(),
+      index: jest.fn(),
+      joinNation: jest.fn(),
+      leaveNation: jest.fn()
+    }
+  }
+}
+const stepper = (fn) => (mock) => fn.next(mock).value
```
Comment (seland) [inline __tests__/src/sagas/nations.js]: Looks like common function used in multiple files, consider extracting.

File: __tests__/src/sagas/nations.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      create: jest.fn(),
+      index: jest.fn(),
+      all: jest.fn(),
+      joinNation: jest.fn(),
+      leaveNation: jest.fn()
+    }
+  }
+}));
+
+jest.mock('react-native-config', () => ({
```
Comment (seland) [inline __tests__/src/sagas/nations.js]: Mocks look the same in several files, convention for jest is to have mocks separated from tests in __mocks__ directory (check out this [link](https://facebook.github.io/jest/docs/en/manual-mocks.html)).

File: __tests__/src/sagas/profile.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import { call, put, select, takeEvery } from 'redux-saga/effects'
+import { cloneableGenerator } from 'redux-saga/utils'
+import watchProfileUpdate, { updateProfile, getProfile, getProfileState } from '../../../src/sagas/profile'
+import { REQUEST_PROFILE_UPDATE, DONE_USER_EDITING, SET_USER_PROFILE, REQUEST_GET_PROFILE, CANCEL_USER_EDITING } from '../../../src/actions/profile'
+import { getPangeaLibrary } from '../../../src/services/container'
+
+jest.mock('BITNATION-Pangea-libs', () => ({
```
Comment (seland) [inline __tests__/src/sagas/profile.js]: The same for mocks.

File: __tests__/src/sagas/profile.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+const pangeaLibrary = {
+  profile: {
+    profile: {
+      setProfile: jest.fn(),
+      getProfile: jest.fn()
+    }
+  }
+}
+const stepper = (fn) => (mock) => fn.next(mock).value
```
Comment (seland) [inline __tests__/src/sagas/profile.js]: The same as for stepper nations reducer tests.

File: __tests__/src/reducers/nations.js (hunk 1 above)
Comment (SeniorMobileDeveloper) [reply __tests__/src/reducers/nations.js]: DONE_FETCH_NATIONS is tested using the action creator doneNationCreation().
Please have a look at the line 33.

Comment (seland) [reply __tests__/src/reducers/nations.js]: Sorry, I mean not for action itself, but for handling of action by reducer.

Comment (SeniorMobileDeveloper) [reply __tests__/src/reducers/nations.js]: const state2 = reducer(state, doneNationCreation())
expect(state2).toEqual({
	...state,
	nation: nationData,
	creatingNation: null,
	inProgress: false
})

I have tested the action handling obviously. Sorry, I don't understand what you are meaning.

Comment (seland) [reply __tests__/src/reducers/nations.js]: I meant done fetch, but not done creation. Please check the [link](https://coveralls.io/builds/15410465/source?filename=src%2Freducers%2Fnations.js#L65) from coveralls.io

Comment (SeniorMobileDeveloper) [reply __tests__/src/reducers/nations.js]: Okay, I will add them now.

**Pull request merged**